components/config/opts.rs

```diff
@@ -657,11 +657,10 @@ pub fn from_cmdline_args(args: &[String]) -> ArgumentParsingResult {
     }
 
     let cwd = env::current_dir().unwrap();
-    let homepage_pref = PREFS.get("shell.homepage");
     let url_opt = if !opt_match.free.is_empty() {
         Some(&opt_match.free[0][..])
     } else {
-        homepage_pref.as_string()
+        None
```

Comment:
> Forgive my newbiness

I've been through this too ^^

> but I'm having some problem with assigning None on a matching statement. I couldn't search up exactly why it does not work. I was wondering if you could give me a pointer for the syntax.

Sure. The issue here is that the variable `url` is a ServoUrl. So the first branch of the `match` does return a ServoUrl. But the `None` branch returns a `Option<ServoUrl>`.

Before your PR, we had a `url_opt` (optional url: `Option<ServoUrl>`) that was transformed to url(`ServoUrl`). Because we didn't want url to be optional. Now we want. So we want to keep url optional.

So in this function, we don't want to `unwrap` the url (which mean moving from `Option<X>` to `X`), we want to keep it optional.

So this block of code your working on, was doing 2 things: 1) calling `parse_url_or_filename` to make sure it's a valid url and that a file path is transformed into a url, and 2) unwrap the url to enforce it (not make it optional).

Now we only want 1) to happen.

Let me know if you need more info. You can also ping me on irc (paul).